core/processor/src/main/java/io/quarkus/annotation/processor/generate_doc/DocGeneratorUtil.java

```diff
@@ -195,4 +217,8 @@ static String join(String delim, Iterator<String> it) {
     static String hyphenate(String orig) {
         return join("-", lowerCase(camelHumpsIterator(orig)));
     }
+
+    static String hyphenateEnumValue(String orig) {
```

Comment:
We should probably skip these cases: 
- an enum with a `@DefaultConverter`  e.g a`Protocol` enum here https://github.com/quarkusio/quarkus/blob/master/extensions/vertx-web/runtime/src/main/java/io/quarkus/vertx/web/runtime/ServerSslConfig.java#L28-L30 and its converter here  
https://github.com/quarkusio/quarkus/blob/master/core/runtime/src/main/java/io/quarkus/runtime/configuration/ssl/ProtocolConverter.java 
- an config item with a custom `@ConvertWith()`, I do not think we have any example from an extension but this is possible. Example usage of `@ConvertWith()` can be found here https://github.com/quarkusio/quarkus/blob/master/core/test-extension/runtime/src/main/java/io/quarkus/extest/runtime/config/TestRunTimeConfig.java#L72

Reply:
And we need to check if it's OK with the Hibernate Search enums as we have a MP config `parse` method there.

Reply:
> And we need to check if it's OK with the Hibernate Search enums as we have a MP config parse method there.

Well, they have an `of(String)` method, as I mentioned, but it just maps `FOO_BAR` to `foo-bar`. So we're good, _except_ that we can't assume all custom converters will do the same thing. I suggest we presume enums with custom converters map that way, rather than defaulting to the more useless type of `String`, no?

Otherwise we need to special-case known enums, but that's sad.

Reply:
> an enum with a @DefaultConverter e.g aProtocol enum here https://github.com/quarkusio/quarkus/blob/master/extensions/vertx-web/runtime/src/main/java/io/quarkus/vertx/web/runtime/ServerSslConfig.java#L28-L30 and its converter here
https://github.com/quarkusio/quarkus/blob/master/core/runtime/src/main/java/io/quarkus/runtime/configuration/ssl/ProtocolConverter.java

I don't get it. How is `Protocol` even involved? It's a `List<String>`.

Reply:
`Protocol` was just an example of cases where you'd want to bail out (hence the  `@DefaultConverter`) from the usage of hyphenated enum converter and use a `custom MP config converter`.